barretenberg/cpp/src/barretenberg/vm2/tracegen/opcodes/get_env_var.test.cpp

```diff
@@ -0,0 +1,342 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+#include "barretenberg/vm2/common/aztec_constants.hpp"
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/common/opcodes.hpp"
+#include "barretenberg/vm2/constraining/flavor_settings.hpp"
+#include "barretenberg/vm2/constraining/full_row.hpp"
+#include "barretenberg/vm2/testing/instruction_builder.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/execution_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::tracegen {
+namespace {
+
+using enum ::bb::avm2::WireOpCode;
+
+class GetEnvVarTracegenTest : public ::testing::Test {
```

Comment:
I don't feel _too_ strongly in this case, but in general TEST_F should be avoided in favour of TEST unless it gives some real benefit.

Personally I don't mind some repetition in tests: https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html

Reply:
What you can consider is using `TEST_P` for parametrized tests. We have a few cases.